modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItJoinTest.java

```diff
@@ -1073,4 +1077,36 @@ private static Stream<Arguments> joinTypes() {
 
         return types;
     }
+
+    @Test
+    public void testNaturalJoinTypeMismatch() {
```

Comment:
I propose to mention that is temporary issue and mention a ticket here which should fix the issue.

Reply:
Done.